cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -1383,5 +1400,188 @@ TEST(Substrait, JoinPlanInvalidKeys) {
+#ifdef _WIN32
+  GTEST_SKIP() << "ARROW-16392: Substrait File URI not supported for Windows";
+#else
+  ExtensionSet ext_set;
+  compute::ExecContext exec_context;
+
+  auto dummy_schema = schema({field("a", int32()), field("b", int32())});
+  auto table = TableFromJSON(dummy_schema, {R"([
+    [1, 1],
+    [3, 4]
+                        ])",
+                                            R"([
+    [0, 2],
+    [1, 3],
+    [4, 1],
+    [3, 1],
+    [1, 2]
+                        ])",
+                                            R"([
+    [2, 2],
+    [5, 3],
+    [1, 3]
+                        ])"});
+  // Note:: adding `/` to support the currently supported file read format from file
+  // system for Substrait
+  const std::string path = "/testing.parquet";
+
+  EXPECT_OK_AND_ASSIGN(auto filesystem,
+                       fs::internal::MockFileSystem::Make(fs::kNoTime, {}));
+
+  EXPECT_EQ(WriteParquetData(path, filesystem, table), true);
+
+  auto format = std::make_shared<arrow::dataset::ParquetFileFormat>();
+
+  std::vector<fs::FileInfo> files;
+  ASSERT_OK_AND_ASSIGN(auto f_file, filesystem->GetFileInfo(path));
+  files.push_back(std::move(f_file));
+
+  ASSERT_OK_AND_ASSIGN(auto ds_factory, dataset::FileSystemDatasetFactory::Make(
+                                            std::move(filesystem), std::move(files),
+                                            std::move(format), {}));
+
+  ASSERT_OK_AND_ASSIGN(auto dataset, ds_factory->Finish(dummy_schema));
+
+  auto options = std::make_shared<dataset::ScanOptions>();
+  options->projection = compute::project({}, {});
+
+  auto scan_node_options = dataset::ScanNodeOptions{dataset, options};
+
+  arrow::AsyncGenerator<util::optional<compute::ExecBatch> > sink_gen;
+
+  auto sink_node_options = compute::SinkNodeOptions{&sink_gen};
+
+  auto scan_declaration = compute::Declaration({"scan", scan_node_options});
+  auto sink_declaration = compute::Declaration({"sink", sink_node_options});
+
+  auto declarations =
+      compute::Declaration::Sequence({scan_declaration, sink_declaration});
+
+  ASSERT_OK_AND_ASSIGN(auto plan, compute::ExecPlan::Make(&exec_context));
+  ASSERT_OK_AND_ASSIGN(auto decl, declarations.AddToPlan(plan.get()));
+
+  ASSERT_OK(decl->Validate());
+
+  std::shared_ptr<arrow::RecordBatchReader> sink_reader = compute::MakeGeneratorReader(
+      dummy_schema, std::move(sink_gen), exec_context.memory_pool());
+
+  ASSERT_OK(plan->Validate());
+  ASSERT_OK(plan->StartProducing());
+
+  std::shared_ptr<arrow::Table> response_table;
+
+  ASSERT_OK_AND_ASSIGN(response_table,
+                       arrow::Table::FromRecordBatchReader(sink_reader.get()));
+
+  ASSERT_OK_AND_ASSIGN(auto serialized_rel,
+                       SerializeRelation(scan_declaration, &ext_set));
+  ASSERT_OK_AND_ASSIGN(auto deserialized_decl,
+                       DeserializeRelation(*serialized_rel, ext_set));
+
+  arrow::AsyncGenerator<util::optional<compute::ExecBatch> > des_sink_gen;
```

Comment:
What is the `des` prefix for?  Can you use a whole word?

Reply:
`des` was meant to represent `deserialized`. I will update it's usage.